src/Promitor.Tests.Unit/Configuration/RuntimeConfigurationUnitTest.cs

```diff
@@ -323,7 +340,23 @@ public async Task RuntimeConfiguration_HasNoHttpPort_UsesDefault()
             // Assert
             Assert.NotNull(runtimeConfiguration);
             Assert.NotNull(runtimeConfiguration.Server);
-            Assert.Equal(Defaults.Server.HttpPort, runtimeConfiguration.Server.HttpPort);
+            Assert.Equal(DefaultsCore.Server.HttpPort, runtimeConfiguration.Server.HttpPort);
+        }
+
+        [Fact]
+        public async Task RuntimeConfiguration_HasNoMaxDegreeOfParallelism_UsesDefault()
+        {
+            // Arrange
+            var configuration = await RuntimeConfigurationGenerator.WithServerConfiguration(null)
```

Comment:
That value is an int so null won't work. Rather than making it nullable, I believe the current behavior is better since the objective is to pass null to indicate use no server configuration at all. (Yes, it is overloading the httpPort value to accomplish this, but if that isn't desirable then I'd split this into different methods.)